fwupdate.py

```diff
@@ -28,25 +36,65 @@ def parse_args() -> argparse.Namespace:
     parser.add_argument(
         "img",
         type=str,
-        help="Mandatory argument of type string for IMG file (make sure the path to this file was mounted when running the pod via -v /host/img:/container/img).",
+        nargs="?",
+        default=None,
+        help=f'IMG file with firmware. For path names, make sure the path is reachable from inside the container (e.g. run `podman -v /:/host`). This can also be a HTTP/HTTPS URL. The special words "uboot" (for "{DEFAULT_IMG_UBOOT}" and "uefi" (for "{DEFAULT_IMG_UEFI}") are supported. The default depends on "--boot-device" and is "uboot" (for "primary") or "uefi" (for "secondary").',
     )
     parser.add_argument(
         "--dev",
         type=str,
         default="eno4",
         help="Optional argument of type string for device. Default is 'eno4'.",
     )
+    parser.add_argument(
+        "--prompt",
+        action="store_true",
+        help="If set, start DHCP/TFTP/HTTP services and wait for the user to press ENTER. This can be used to manually setup the host side serving the firmware.",
+    )
+    parser.add_argument(
+        "-B",
+        "--boot-device",
+        choices=["primary", "secondary"],
+        default="secondary",
+        help='Select primary or secondary boot device. Defaults to "secondary".',
+    )
 
     args = parser.parse_args()
-    if not os.path.exists(args.img):
-        print(f"Couldn't read img file {args.img}")
-        raise Exception("Invalid path to omg provided")
 
     return args
 
 
+def prepare_image(boot_device: str, img: typing.Optional[str]) -> str:
+    if not img:
+        if boot_device == "primary":
+            img = "uboot"
+        else:
+            img = "uefi"
+
+    if img == "uboot":
+        img = DEFAULT_IMG_UBOOT
+    elif img == "uefi":
+        img = DEFAULT_IMG_UEFI
+
+    if img.startswith("http://") or img.startswith("https://"):
+        img2 = "/tmp/fwupdate.img"
+        logger.info(f"downloading {repr(img)} to {repr(img2)}.")
+        host.local.run(
+            ["curl", "-k", "-L", "-o", img2, img],
```

Comment:
with https, most likely you need the insecure flag for curl

Reply:
yes, that is the `-k`.

Isn't it?

Reply:
Ah you are right